packages/api/cms-api/src/generator/generate-crud.ts

```diff
@@ -462,7 +462,16 @@ function generateService({ generatorOptions, metadata }: { generatorOptions: Cru
     const { classNameSingular, fileNameSingular, classNamePlural } = buildNameVariants(metadata);
     const { hasPositionProp, positionGroupProps } = buildOptions(metadata, generatorOptions);
 
-    const positionGroupType = positionGroupProps.length ? `{ ${positionGroupProps.map((prop) => `${prop.name}: ${prop.type}`).join(",")} }` : false;
+    const positionGroupType = positionGroupProps.length
+        ? `{ ${positionGroupProps
+              .map(
+                  (prop) =>
+                      `${prop.name}${prop.nullable ? `?` : ``}: ${
+                          [ReferenceType.EMBEDDED, ReferenceType.SCALAR].includes(prop.reference) ? prop.type : "string"
```

Comment:
@johnnyomair @nsams I think we don't want to support `OneToMany` and `ManyToMany` fields in `groupByFields`. Should I throw an error?